src/main/c/seasocks/Connection.h

```diff
@@ -227,6 +229,8 @@ class Connection : public WebSocket {
     };
     State _state;
 
+    std::shared_ptr<void> _userdata;
```

Comment:
shared_ptr to void seems a little weird. It took me a while to remember that the destructor of the actual underlying object is type-erased, else I was worried that we'd need a virtual base class.

Maybe I'm being a bit funny about this, but I think I mildly prefer a virtual base class here. But I can be persuaded :)

Reply:
I don't think a virtual base class is a good idea, as it would result in having to modify existing classes in order to use them here, which might be hard or even impossible (eg if you only have a opaque c pointer).